src/air/constraints/boundary.rs

```diff
@@ -105,14 +114,14 @@ impl<F: IsField> BoundaryConstraints<F> {
         primitive_root: &FieldElement<F>,
         col: usize,
     ) -> Polynomial<FieldElement<F>> {
-        let mut zerofier = Polynomial::new_monomial(FieldElement::<F>::one(), 0);
-        for step in self.steps(col).into_iter() {
-            let binomial = Polynomial::new(&[-primitive_root.pow(step), FieldElement::<F>::one()]);
-            // TODO: Implement the MulAssign trait for Polynomials?
-            zerofier = zerofier * binomial;
-        }
-
-        zerofier
+        self.steps(col).into_iter().fold(
+            Polynomial::new_monomial(FieldElement::<F>::one(), 0),
```

Comment:
Im not sure the impact of this, but maybe we can save ourselves from multiplying the polynomial one by starting the fold at the first element instead

Reply:
Oh, I see the exponents are a bit more complicated than that. But I still think something along those lines may be useful.